### PR TITLE
Add allen cahn sota

### PR DESCRIPTION
**Pull request by xusuyong**
### PR types

New features
### PR changes

Others

### CODE REVIEW

File: ppsci/loss/mtl/ntk.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+                ntk_value.append(
+                    paddle.sqrt(
+                        paddle.sum(grad.detach() ** 2),
+                    )
+                )
+
+        ntk_sum += paddle.sum(paddle.stack(ntk_value, axis=0))
+        ntk_weight = [(ntk_sum / x) for x in ntk_value]
+
+        return ntk_weight
```
Comment (HydrogenSulfate) [inline ppsci/loss/mtl/ntk.py:60]: 感觉这里的逻辑不太对，K矩阵的形状应该是`[batch, batch]`，`K[i][j]`表示第i个样本的loss对模型参数的梯度向量，与第j个样本对模型参数的梯度向量的点积。从结果上来看，`K.trace()`和上述结果应该是不等价的，因为平方再求和，与求和再平方是不等价的。

File: ppsci/loss/mse.py
Comment (HydrogenSulfate) [inline ppsci/loss/mse.py:1]: CausalMSELoss的权重计算过程应该需要把NTK考虑进来，因为源论文的causalmseloss在NTK是否开启有两种实现方式

**Review by HydrogenSulfate — CHANGES_REQUESTED** [review]

File: examples/allen_cahn/conf/allen_cahn_defalut_ntk.yaml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  iters_per_epoch: 1000
+  save_freq: 10
+  eval_during_train: true
+  eval_freq: 1
+  lr_scheduler:
+    epochs: ${TRAIN.epochs}
+    iters_per_epoch: ${TRAIN.iters_per_epoch}
+    learning_rate: 1.0e-3
+    gamma: 0.9
+    decay_steps: 2000
```
Comment (HydrogenSulfate) [inline examples/allen_cahn/conf/allen_cahn_defalut_ntk.yaml:63]: 这里应该是5000

**Review by HydrogenSulfate — CHANGES_REQUESTED** [review]

**Review by HydrogenSulfate — DISMISSED** [review]
LGTM

**Review by HydrogenSulfate — APPROVED** [review]
LGTM，辛苦提交PR
